packages/@aws-cdk/aws-stepfunctions/lib/states/map.ts

```diff
@@ -172,10 +212,10 @@ export class Map extends State implements INextable {
       ResultPath: renderJsonPath(this.resultPath),
       ...this.renderNextEnd(),
       ...this.renderInputOutput(),
-      ...this.renderParameters(),
+      ...this.renderItemSelector(),
       ...this.renderResultSelector(),
       ...this.renderRetryCatch(),
-      ...this.renderIterator(),
+      ...this.renderItemProcessor(),
```

Comment:
We should likely still render both. If it's an error to render both, then we should throw an error if we detect that both have been specified.